extensions-contrib/opentelemetry-extensions/src/test/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReaderTest.java

```diff
@@ -0,0 +1,276 @@
+  public static final String RESOURCE_ATTRIBUTE_ENV = "env";
+  public static final String RESOURCE_ATTRIBUTE_VALUE_DEVEL = "devel";
+
+  public static final String INSTRUMENTATION_LIBRARY_NAME = "mock-instr-lib";
+  public static final String INSTRUMENTATION_LIBRARY_VERSION = "1.0";
+
+  public static final String METRIC_ATTRIBUTE_COLOR = "color";
+  public static final String METRIC_ATTRIBUTE_VALUE_RED = "red";
+
+  public static final String METRIC_ATTRIBUTE_FOO_KEY = "foo_key";
+  public static final String METRIC_ATTRIBUTE_FOO_VAL = "foo_value";
+
+  private final MetricsData.Builder metricsDataBuilder = MetricsData.newBuilder();
+
+  private final Metric.Builder metricBuilder = metricsDataBuilder.addResourceMetricsBuilder()
+          .addInstrumentationLibraryMetricsBuilder()
+          .addMetricsBuilder();
+
+  private final DimensionsSpec dimensionsSpec = new DimensionsSpec(ImmutableList.of(
+          new StringDimensionSchema("descriptor." + METRIC_ATTRIBUTE_COLOR),
+          new StringDimensionSchema("descriptor." + METRIC_ATTRIBUTE_FOO_KEY),
+          new StringDimensionSchema("custom." + RESOURCE_ATTRIBUTE_ENV),
+          new StringDimensionSchema("custom." + RESOURCE_ATTRIBUTE_COUNTRY)
+  ), null, null);
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Before
+  public void setUp()
+  {
+    metricsDataBuilder
+            .getResourceMetricsBuilder(0)
+            .getResourceBuilder()
+            .addAttributes(KeyValue.newBuilder()
+                    .setKey(RESOURCE_ATTRIBUTE_COUNTRY)
+                    .setValue(AnyValue.newBuilder().setStringValue(RESOURCE_ATTRIBUTE_VALUE_USA)));
+
+    metricsDataBuilder
+            .getResourceMetricsBuilder(0)
+            .getInstrumentationLibraryMetricsBuilder(0)
+            .getInstrumentationLibraryBuilder()
+            .setName(INSTRUMENTATION_LIBRARY_NAME)
+            .setVersion(INSTRUMENTATION_LIBRARY_VERSION);
+
+  }
+
+  @Test
+  public void testSumWithAttributes()
+  {
+    metricBuilder
+            .setName("example_sum")
+            .getSumBuilder()
+            .addDataPointsBuilder()
+            .setAsInt(6)
+            .setTimeUnixNano(TIMESTAMP)
+            .addAttributesBuilder() // test sum with attributes
+            .setKey(METRIC_ATTRIBUTE_COLOR)
+            .setValue(AnyValue.newBuilder().setStringValue(METRIC_ATTRIBUTE_VALUE_RED).build());
+
+    MetricsData metricsData = metricsDataBuilder.build();
+
+    CloseableIterator<InputRow> rows = new OpenTelemetryMetricsProtobufReader(
+            dimensionsSpec,
+            new ByteEntity(metricsData.toByteArray()),
+            "metric.name",
+            null,
+            "descriptor.",
+            "custom."
+    ).read();
+
+    List<InputRow> rowList = new ArrayList<>();
+    rows.forEachRemaining(rowList::add);
+    Assert.assertEquals(1, rowList.size());
+
+    InputRow row = rowList.get(0);
+    Assert.assertEquals(4, row.getDimensions().size());
```

Comment:
nit, we check for 4 dimensions, but only verify 4 of them, can we add the 4th one?